packages/core/src/runner/helper.ts

```diff
@@ -76,27 +78,25 @@ export function createPipeline(components: PipcookComponentResult[], self: Pipco
   const firstComponent = components[0];
   firstComponent.status = 'running';
   logCurrentExecution(firstComponent, logType);
-  const insertParams = {
+  const insertParams: InsertParams = {
     pipelineId: self.pipelineId,
     modelDir: path.join(self.logDir, 'model'),
     dataDir: path.join(self.logDir, 'data')
   };
   const firstObservable = firstComponent.observer(null, self.latestModel, insertParams);
   self.updatedType = firstComponent.returnType;
 
-  const flatMapArray: any = [];
+  const flatMapArray: PipcookComponentOperator[] = [];
```

Comment:
How about using `reduce` function to remove the temp assignment?

Reply:
`reduce` is more elegant in coding, but less readable than a simple loop, It is more suitable when the result has the same type as the items